engine/crates/engine-v2/schema/src/builder/ids.rs

```diff
@@ -0,0 +1,179 @@
+//! This module is responsible for mapping identifiers between federated schema and the engine v2
+//! schema when the mapping is not 1:1.
+//!
+//! As of 3b56f12c95d334ce6cb46f4f8654ce531a69f975, this only happens when @inaccessible items
+//! are removed.
+
+use crate::ids::*;
+
+macro_rules! mapped_ids {
+    ($($name:ident : $from:ty => $to:ty)*) => {
```

Comment:
Probably you could have avoided the macro stuff by making it generic `IdMapper<Origin: Into<usize>, Target: From<usize>>` and just instantiating the right id mappers in the schema builder. A bit late as a comment though^^

Reply:
Hmm true